### ISSUE TITLE
Static Posts?

### ISSUE THREAD

**Issue by benhamill** [Plume-org/Plume#23]
It would be nice if the server software didn't have to make a DB call when someone went to a post's page, but instead rendered as much as possible to static HTML. Having the advantages of static-site generators like Jekyll and friends with the nice-ness of web publishing and federation would be… potentially awesome.

**Comment by BaptisteGelez** [comment]
Yes, in the future I think the content of the posts will be stored in real files, allowing versionning and patching them (for further edits).

**Comment by Zykino** [comment]
If I could write my posts in Markdown and they are kept in .md somewhere that would be perfect :)

**Comment by southerntofu** [comment]
I would be interested to run plume as a social frontend to [zola](https://getzola.org) which i use for my blog. Abstracting over the storage backend would be ideal, but if that's an approach that is not of interest to you, maybe a simple hook system could do it?

If we could have a hook environment variable pointing to an executable on the system, that would take a few well-defined (and stable across versions) arguments - such as "user add" or "post remove" - with Plume's data structure sent to STDIN as JSON or some other standard serialization format, i think that could do the trick. I believe it would require to version the internal data structures to ensure 3rd party tools quit gracefully in case of version mismatch, instead of failing horribly in unpredictable ways.

There's probably many different ways to attack this problem so i'm interested to hear other thoughts on this topic.

**Comment by elegaanz** [comment]
For the moment, a solution to integrate a static blog in the Fediverse could be squs : github.com/elegaanz/squs/

It is a Plume fork I made a few months ago, that reads the RSS feed of a blog and makes a bridge between the Fediverse and the static blog, providing a script to display comments, like Disqus does.

It may be broken tho, I didn't tested it in a long time (and it is based on a quite old version of Plume).
